### ISSUE TITLE
decoding sklearn models using jsonpickle fails

### ISSUE THREAD

**Issue by bharatl** [jsonpickle/jsonpickle#147]
I am trying to *encode* and *decode* __scikit-learn__ ML model using __jsonpickle__. Encoding works fine, but when I try to decode it throws me an error saying  
```
AttributeError("'dict' object has no attribute '__name__'",) 
{'py/object': '__builtin__.builtin_function_or_method'}
```
here is the sample code 
```
from sklearn import svm
import jsonpickle
from sklearn.datasets import load_iris
from sklearn.neighbors import KNeighborsClassifier

if __name__ == '__main__':
    iris = load_iris()
    X = iris.data
    y = iris.target
    knn = KNeighborsClassifier(n_neighbors=5)
    knn.fit(X, y)
    data = jsonpickle.encode(knn)
    pklload = jsonpickle.decode(data)
```
ERROR
```
    if f == tags.NEWOBJ or f.__name__ == '__newobj__':
AttributeError: 'dict' object has no attribute '__name__'
```
Can Jsonpickle be used for serializing sklearn model's  to Json?

**Comment by bharatl** [comment]
I am on python 3.5 and 
>>> jsonpickle.__version__
'0.9.3'
>>>

**Comment by anntzer** [comment]
In fact even numpy arrays fail with the same error (python 3.5.2):
```
$ python -c 'import jsonpickle, numpy as np; print(jsonpickle.loads(jsonpickle.dumps(np.array([1]))))'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.5/site-packages/jsonpickle/__init__.py", line 152, in decode
    return unpickler.decode(string, backend=backend, keys=keys)
  File "/usr/lib/python3.5/site-packages/jsonpickle/unpickler.py", line 26, in decode
    return context.restore(backend.decode(string), reset=reset)
  File "/usr/lib/python3.5/site-packages/jsonpickle/unpickler.py", line 119, in restore
    value = self._restore(obj)
  File "/usr/lib/python3.5/site-packages/jsonpickle/unpickler.py", line 159, in _restore
    return restore(obj)
  File "/usr/lib/python3.5/site-packages/jsonpickle/unpickler.py", line 176, in _restore_reduce
    if f == tags.NEWOBJ or f.__name__ == '__newobj__':
AttributeError: 'dict' object has no attribute '__name__'
```

**Comment by crawles** [comment]
I also observe this issue

**Comment by davvid** [comment]
I'll add something to the README that makes this easier to discover.

The latest jsonpickle has much better numpy support (especially so on python3) thanks to @EelcoHoogendoorn 

The magic incantation you need to add to your script is this:
```
import jsonpickle.ext.numpy as jsonpickle_numpy
jsonpickle_numpy.register_handlers()
```

and the issues observed here will all be resolved.

**Comment by mohitbadwal** [comment]
I don't understand why this issue is closed , even after trying the method mentioned by @davvid I still get the same error for KNeighborsClassifier

`
if f == tags.NEWOBJ or f.__name__ == '__newobj__':
AttributeError: 'dict' object has no attribute '__name__'
`

**Comment by davvid** [comment]
If you're getting that error it means you're running an older version of jsonpickle, as that line of code does not look that way anymore.  Did you try the code in `master`?  Please submit a pull request with a test case if you're still seeing issues.

**Comment by davvid** [comment]
Since you're using old code I'm assuming that you're installing code from pypi.  I just tagged and released `v0.9.6` to pypi so if you upgrade you should at least get the latest code.  Please feel free to open a new issue if you're still seeing problems after the upgrade.
